src/main/java/de/retest/recheck/auth/RetestAuthentication.java

```diff
@@ -80,6 +81,11 @@ private JWTVerifier getJwtVerifier() {
 			final UrlJwkProvider provider = new UrlJwkProvider( URI.create( CERTS_URL ).toURL() );
 			final PublicKey publicKey = provider.get( PUBLIC_KEY_ID ).getPublicKey();
 			return JWT.require( Algorithm.RSA256( (RSAPublicKey) publicKey, null ) ).acceptLeeway( 3 ).build();
+		} catch (SigningKeyNotFoundException e ) {
+			RuntimeException offline = new UnableToAuthenticateOfflineException( e );
+			log.error( "Exception retrieving signing key: ", offline );
+			handler.loginFailed( offline );
+			throw offline;
```

Comment:
Just stepping by: Your formatter / save actions / cleanup seems to be off (again).

```suggestion
		} catch ( SigningKeyNotFoundException e ) {
			final RuntimeException offline = new UnableToAuthenticateOfflineException( e );
			log.error( "Exception retrieving signing key: ", offline );
			handler.loginFailed( offline );
			throw offline;
```

Also, there is usually no need to log & re-throw, doing the latter should already result in a corresponding log. Plus: the handler logs as well.

Reply:
<img width="573" alt="Screen Shot 2020-04-15 at 15 04 20  08 29 49" src="https://user-images.githubusercontent.com/1871610/79305268-5bc4d380-7ef3-11ea-967c-8f3845d0e060.png">

Reply:
This is because of Lombok, download the edge version and it should be fixed: https://projectlombok.org/download-edge

Reply:
@roesslerj Have you adressed the comment by @beatngu13 or maybe discussed in a side-conversation? I don't see any changes regarding those comments:

> Also, there is usually no need to log & re-throw, doing the latter should already result in a corresponding log. Plus: the handler logs as well.

Reply:
Sorry, committing the suggesting automatically resolved that conversation...
> Also, there is usually no need to log & re-throw, doing the latter should already result in a corresponding log. Plus: the handler logs as well.
I tested this, and calling `handler.loginFailed(..)` exits the application. But I cannot return nothing, so I rethrow. Logging twice in that situation doesn't hurt I guess.

Reply:
But since we implement the handler in a downstream project, we can only adjust it there...